### PR TITLE
fix(http): check response body text against undefined

### PR DESCRIPTION
**Pull request by domusofsail**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

this._body.toString() could result into "Cannot read property 'toString' of undefined" error

**What is the new behavior?**

Fixes the issue

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Check this._body against undefined too otherwise
this._body.toString() could result into
"Cannot read property 'toString' of undefined" error

### CODE REVIEW

File: modules/@angular/http/src/body.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export abstract class Body {
       return String.fromCharCode.apply(null, new Uint16Array(<ArrayBuffer>this._body));
     }
 
-    if (this._body === null) {
+    if (this._body === null  || typeof this._body === 'undefined') {
```
Comment (DzmitryShylovich) [inline modules/@angular/http/src/body.ts]: `this._body == null`

**Comment by domusofsail** [comment]
I signed it!

**Comment by DzmitryShylovich** [comment]
Pls squash your commits and add a test.

**Comment by domusofsail** [comment]
Test added

**Pull request merged**